src/ansi.rs

```diff
@@ -0,0 +1,1821 @@
+//! Process parsed terminal streams into terminal actions.
+
+use core::convert::TryFrom;
+use core::{iter, str};
+
+use core::time::Duration;
+
+#[cfg(feature = "alloc")]
+use alloc::string::String;
+#[cfg(feature = "alloc")]
+use alloc::vec::Vec;
+
+#[cfg(not(feature = "alloc"))]
+use arrayvec::ArrayVec;
+use log::{debug, trace};
+
+use crate::{Params, ParamsIter, Parser, Perform};
+use core::option::Option::Some;
```

Comment:
Imports should be ordered appropriately:

```suggestion
use core::convert::TryFrom;
use core::{iter, str};
use core::option::Option::Some;
use core::time::Duration;

#[cfg(feature = "alloc")]
use alloc::string::String;
#[cfg(feature = "alloc")]
use alloc::vec::Vec;

#[cfg(not(feature = "alloc"))]
use arrayvec::ArrayVec;
use log::{debug, trace};

use crate::{Params, ParamsIter, Parser, Perform};
```